src/test/java/no/unit/nva/handlers/AddRoleHandlerTest.java

```diff
@@ -134,6 +134,27 @@ public void errorMessageIsLoggedWhenAddRoleHandlerThrowsDatasyncException()
         assertThat(testingAppender.getMessages(), containsString(AddRoleHandler.ERROR_FETCHING_SAVED_ROLE));
     }
 
+    @Test
+    public void handleRequestReturnsBadRequestWhenInputRoleHasNoType()
+        throws InvalidEntryInternalException, IOException {
+        RoleDto roleDto = createSampleRole();
+        ObjectNode objectWithoutType = createInputObjectWithoutType(roleDto);
+
+        InputStream requestInput = createRequestInputStream(objectWithoutType);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+        addRoleHandler.handleRequest(requestInput, outputStream, context);
+        GatewayResponse<Problem> response = GatewayResponse.fromOutputStream(outputStream);
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_BAD_REQUEST)));
+
+        Problem problem = response.getBodyObject(Problem.class);
+        assertThat(problem.getDetail(), is(equalTo(InvalidOrMissingTypeException.MESSAGE)));
```

Comment:
Given the above refactoring, you can split the two asserts into separate tests calling the extracted method